docs/proposals/enable-local-source-code-support.md

```diff
@@ -0,0 +1,301 @@
+
+#### As a Shipwright User I want to build images in Development mode from my local Dockerfile or plain source code
+
+Users should only need to specify the mandatory fields under `spec.source.container` without modifying their existing strategies of choice. Local source code feature should work out of the box with different tools, like Kaniko, BuildKit, Buildpacks, et al.
+
+This will be done via Shipwright/CLI in the form of:
+
+```sh
+shp build create a-build --local-path "/absolute-path/src/github.com/shipwright-io/sample-go/docker-build" --output-image "docker.io/<registry-repository>/img:latest" --strategy "kaniko"
+```
+
+_Note_: The above is just an example on how the CLI might look like.
+
+#### As a Shipwright User I want to ensure reproducible bundle images
+
+We want to provide trust to users when building bundle images with their source code, so that on multiple builds with the same local source code, the bundle image always express the same _SHA_. Subsequent Builds should generate the same image if the source code is unchanged.
+
+#### As a Shipwright User I want to avoid excessive registry storage from Bundle images
+
+We want to allow users to signalize the desire of pruning a bundle image that was used for building their application container image.
+
+Adding `pruneBundle` should ensure a mechanism for deleting `docker.io/<registry-repository>/bundle:latest`.
+
+```yaml
+---
+apiVersion: shipwright.io/v1alpha1
+kind: Build
+metadata:
+  name: a-kaniko-build
+spec:
+  source:
+    container:
+      image: docker.io/<registry-repository>/bundle:latest
+      pruneBundle: true # Disable by default
+  strategy:
+    name: kaniko
+    kind: ClusterBuildStrategy
+  output: ...
+```
+
+The above Build definition should ensure that a step in our [TaskSpec](https://github.com/qu1queee/build/blob/qu1queee/local_source/pkg/reconciler/buildrun/resources/taskrun.go#L59) definition is appended, which will ensure a pruning of the image in the related container registry.
+
+_Note_: Supporting multiple registry providers might be too complicated and different alternatives should be considering for pruning.
+
+#### As a Shipwright/CLI contributor I want to provide a mechanism to support local source code
+
+This is related to the CLI workflow, that should support the bundle approach and maintain the bundle base image. The implementation should be generic enough for other workflows, like OpenShift Build, IBM Cloud Code Engine, to consume it.
+
+### Implementation Details/Notes/Constraints [optional]
+
+There is a prototype for educational purposes that can be tested if desire, see the related [documentation](https://github.com/qu1queee/cli/blob/qu1queee/crud_cmd/README.md). This uses the following two branches:
+
+- Changes in Shipwright/Build, see [branch](https://github.com/qu1queee/build/tree/qu1queee/local_source)
+- Changes in Shipwright/CLI, see [branch](https://github.com/qu1queee/cli/tree/qu1queee/crud_cmd)
+
+### Risks and Mitigations
```

Comment:
There's a risk that when source is pushed without a tag, the target container registry will prune the image before we pull it in our build. How do we mitigate this?

Reply:
I was not aware on that behaviour. Any links? I think we should enforce a tag in CLI.